### PR TITLE
Extend API

### PR DESCRIPTION
**Pull request by martinstark**
Adds the following:

- Ability to fetch the current timestamp including stitched ad durations when viewing VODs. CAUTION: this timestamp may refer to different relative positions on each video start, depending on length of stitched ads
- Ability to fetch the duration including stitched ad durations when viewing VODs. CAUTION: this timestamp may refer to different relative positions on each video start, depending on length of stitched ads

to test, call:

```javascript
player.getCurrentTime(TimeMode.AbsoluteTime);
player.getDuration(TimeMode.AbsoluteTime);
```

If the player is currently playing an ad, by default, the methods would return the position within the ad and the duration of the ad respectively. If the `TimeMode.AbsoluteTime` parameter is added, the methods will instead return the absolute time, ignoring ongoing ads.

### CODE REVIEW

File: src/ts/BitmovinYospacePlayerAPI.ts [hunk 1]
```diff
@@ -49,6 +49,12 @@ export interface BitmovinYospacePlayerAPI extends PlayerAPI {
 
   getCurrentPlayerType(): YospacePlayerType;
 
+  getCurrentAdBreakDuration(): number;
```
Comment (dweinber) [inline src/ts/BitmovinYospacePlayerAPI.ts:52]: Feels like this API isn't necessary as the data is already exposed.`player.ads.getActiveAdBreak()?.duration ?? 0` should be equivalent, or am I missing something?

File: src/ts/BitmovinYospacePlayerAPI.ts [hunk 2]
```diff
@@ -49,6 +49,12 @@ export interface BitmovinYospacePlayerAPI extends PlayerAPI {
...
+
+  getCurrentTimeWithAds(): number;
+
+  getDurationWithAds(): number;
```
Comment (dweinber) [inline src/ts/BitmovinYospacePlayerAPI.ts:56]: I'm inclined to suggest using an optional parameter for the existing `getDuration` API, like `getDuration(withAds = false)`. Thoughts?

File: src/ts/BitmovinYospacePlayerAPI.ts [hunk 3]
```diff
@@ -49,6 +49,12 @@ export interface BitmovinYospacePlayerAPI extends PlayerAPI {
 
   getCurrentPlayerType(): YospacePlayerType;
 
+  getCurrentAdBreakDuration(): number;
+
+  getCurrentTimeWithAds(): number;
```
Comment (dweinber) [inline src/ts/BitmovinYospacePlayerAPI.ts:54]: I'm inclined to suggest using an optional parameter for the existing `getCurrentTime` API, like `getCurrentTime(withAds = false)`. Thoughts?

**Review by dweinber — CHANGES_REQUESTED** [review]
Thanks for the contribution. A few comments to address, and please add a CHANGELOG.md entry - thanks!

File: src/ts/BitmovinYospacePlayerAPI.ts (hunk 1 above)
Comment (martinstark) [reply src/ts/BitmovinYospacePlayerAPI.ts:52]: an explicit method was requested (cc @saravanans-github), I'm happy to remove it if the existing API fulfils the need

File: src/ts/BitmovinYospacePlayerAPI.ts (hunk 2 above)
Comment (martinstark) [reply src/ts/BitmovinYospacePlayerAPI.ts:56]: in slack discussions we opted for explicit methods, but I have no strong feeling here (cc @saravanans-github)

File: src/ts/BitmovinYospacePlayerAPI.ts (hunk 3 above)
Comment (martinstark) [reply src/ts/BitmovinYospacePlayerAPI.ts:54]: in slack discussions we opted for explicit methods, but I have no strong feeling here (cc @saravanans-github)

**Comment by martinstark** [comment]
@dweinber changelog is included in #171 , if that is okay. It could be merged into this branch before merging to develop

**Comment by dweinber** [comment]
Thanks @martinstark, I've started a conversation around existing vs new APIs separately.

Re: Changelog:
I believe what's done in this PR should also be listed by this PR 🙂 (i.e. just the addition of the new APIs should be listed as part of this PR)

**Comment by martinstark** [comment]
> Re: Changelog:
> I believe what's done in this PR should also be listed by this PR 🙂 (i.e. just the addition of the new APIs should be listed as part of this PR)

Sure! Since the PRs are built on top of each other I'll have a small rebase/force-push party to get it sorted once we agree to the finalized API. I'll move the changelog to the appropriate PR when that happens.

**Comment by dweinber** [comment]
Ah ok, missed that. I thought we'd merge this one, and then 171 would be just done against `develop` as well. 

All good then, don't worry about the changelog.

**Comment by martinstark** [comment]
Since I'll likely need to make changes here I'll be doing the rebasing regardless, so I can make sure the changelog is in the correct place as part of that

**Comment by martinstark** [comment]
@dweinber updated the PR

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 - Event loop on pre-roll ad end
 
+### Changed
+
+- added mode argument to getCurrentTime to enable fetching absolute time
+- added mode argument to getDuration to enable fetching absolute duration
```
Comment (dweinber) [inline CHANGELOG.md:17]:
```suggestion
- `mode` argument to `getDuration` to enable fetching absolute duration including ad durations
```

**Review by dweinber — APPROVED** [review]
Code looks good, thanks for the updates!

**Pull request merged**